art/detection/subsetscanning/scanningops.py

```diff
@@ -0,0 +1,144 @@
+
+        for elem_indx in range(0, number_of_elements):
+            #sort all the range maxes
+            if image_to_node:
+                # collect ranges over images(rows)
+                arg_sort_max = np.argsort(pvalues[:, elem_indx, 1])
+                #arg_sort_min = np.argsort(pvalues[:,e,0]) #collect ranges over images(rows)
+                completely_included = np.searchsorted(
+                    pvalues[:, elem_indx, 1][arg_sort_max], alpha_thresholds, side='right')
+            else:
+                # collect ranges over nodes(columns)
+                arg_sort_max = np.argsort(pvalues[elem_indx, :, 1])
+                #arg_sort_min = np.argsort(pvalues[elem_indx,:,0])
+
+                completely_included = np.searchsorted(
+                    pvalues[elem_indx, :, 1][arg_sort_max], alpha_thresholds, side='right')
+
+            #print('complete included shape', completely_included.shape)
+            # should be num elements by num thresh
+            unsort_priority[elem_indx, :] = completely_included
+
+        # want to sort for a fixed thresh (across?)
+        arg_sort_priority = np.argsort(-unsort_priority, axis=0)
+
+        best_score_so_far = -10000
+        best_alpha = -2
+
+        alpha_count = 0
+        for alpha_threshold in alpha_thresholds:
+
+            # score each threshold by itself, cumulating priority,
+            # cumulating count, alpha stays same.
+            alpha_v = np.ones(number_of_elements)*alpha_threshold
+
+            # may need to reverse this?
+            n_alpha_v = np.cumsum(
+                unsort_priority[:, alpha_count][arg_sort_priority][:, alpha_count])
+            count_increments_this = np.ones(number_of_elements)*size_of_given
+            n_v = np.cumsum(count_increments_this)
+
+            vector_of_scores = score_function(n_alpha_v, n_v, alpha_v)
+
+            best_score_for_this_alpha_idx = np.argmax(vector_of_scores)
+            best_score_for_this_alpha = vector_of_scores[best_score_for_this_alpha_idx]
+
+            if best_score_for_this_alpha > best_score_so_far:
+                best_score_so_far = best_score_for_this_alpha
+                best_size = best_score_for_this_alpha_idx + 1  # not sure 1 is needed?
+                best_alpha = alpha_threshold
+                best_alpha_count = alpha_count
+            alpha_count = alpha_count + 1
+
+        # after the alpha for loop we now have best score, best alpha, size of best subset,
+        # and alpha counter use these with the priority argsort to reconstruct the best subset
+
+        unsort = arg_sort_priority[:, best_alpha_count]
+
+        subset = np.zeros(best_size).astype(int)
+        for loc in range(0, best_size):
+            subset[loc] = unsort[loc]
+
+        return(best_score_so_far, subset, best_alpha)
+
+    @staticmethod
+    def single_restart(pvalues, a_max, indices_of_seeds, image_to_node, score_function):
```

Comment:
Could you please add a docstring describing what the function does, its arguments and return similar to the other function docstrings in ART?